### PR TITLE
Support special words

### LINKED ISSUE #2084: use $ as the operation rename for reserved word.
If we have an operation name delete, currently, we will rename it as deleteOperation because of the reserved word. 
But we should rename it as $delete instead. see comment https://github.com/Azure/autorest.typescript/pull/2080#discussion_r1369876176

### LINKED ISSUE #2214: special words in model properties, models, operations, parameters,
TypeSpec link: https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/special-words/main.tsp 

Should consideration this issue https://github.com/Azure/autorest.typescript/issues/2084

### LINKED ISSUE #2352: Operation option interface is changed from DataProductsGetOptionalParams to DataProductsGetOptions

### PR DESCRIPTION
**Pull request by qiaozha**
fix #2214
fix #2352
fix #2084

### CODE REVIEW

File: packages/typespec-ts/test/modularIntegration/generated/encode/bytes/src/classic/header/index.ts [hunk 1]
```diff
@@ -9,37 +9,32 @@ import {
   headerBase64urlArray,
 } from "../../api/header/index.js";
 import {
-  HeaderDefaultOptions,
-  HeaderBase64Options,
-  HeaderBase64urlOptions,
-  HeaderBase64urlArrayOptions,
+  DefaultOptions,
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/generated/encode/bytes/src/classic/header/index.ts:12]: need to figure out why this changed.

File: packages/typespec-ts/test/modularIntegration/generated/scalar/src/ScalarClient.ts [hunk 2]
```diff
@@ -3,13 +3,13 @@
 
 import { Pipeline } from "@azure/core-rest-pipeline";
 import {
-  getStringOperations,
-  StringOperations,
-} from "./classic/string/index.js";
+  getStringModelOperations,
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/generated/scalar/src/ScalarClient.ts:6]: why is this change here ?

File: packages/typespec-ts/test/modularIntegration/generated/server/path/multiple/src/MultipleClient.ts [hunk 3]
```diff
@@ -21,8 +21,8 @@ export class MultipleClient {
   /** The pipeline used by this client to make requests */
   public readonly pipeline: Pipeline;
 
-  constructor(endpoint: string, options: MultipleClientOptions = {}) {
-    this._client = createMultiple(endpoint, options);
+  constructor(endpointParam: string, options: MultipleClientOptions = {}) {
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/generated/server/path/multiple/src/MultipleClient.ts:24]: feel like this change shouldn't be here.

File: packages/typespec-test/test/batch_modular/generated/typespec-ts/review/batch.api.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ export type AccessScope = string;
 
 // @public
 export interface AccountListSupportedImagesResult {
-    "odata.nextLink"?: string;
+    odataNextLink?: string;
```
Comment (qiaozha) [inline packages/typespec-test/test/batch_modular/generated/typespec-ts/review/batch.api.md:17]: we should not normalize the name here.

**Review by MaryGao — APPROVED** [review]
LGTM

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 5]
```diff
@@ -1160,6 +1154,9 @@ export function serializeRequestValue(
         return `${clientValue} as any`;
       }
     default:
+      if (clientValue === "constructorParam") {
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:1157]: is this the only case? do we need to do this for other xxxParam?

Comment (qiaozha) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:1157]: as offline discussed, this is the only case for now.
